src/providers/twitch/eventsub/MessageBuilder.cpp

```diff
@@ -1,8 +1,40 @@
 #include "providers/twitch/eventsub/MessageBuilder.hpp"
 
+#include "common/Literals.hpp"
 #include "messages/Message.hpp"
 #include "messages/MessageBuilder.hpp"
 
+namespace {
+
+using namespace chatterino;
+using namespace chatterino::eventsub;
+using namespace chatterino::literals;
+
+/// <MODERATOR> turned {on/off} <MODE> mode. [<DURATION>]
+void makeModeMessage(EventSubMessageBuilder &builder,
+                     const lib::payload::channel_moderate::v2::Event &event,
+                     const QString &mode, bool on, const QString &duration = {})
+{
+    QString text;
+
+    builder.appendUser(event.moderatorUserName, event.moderatorUserLogin, text);
+    builder.emplaceSystemTextAndUpdate(u"turned"_s, text);
+    QString op = on ? u"on"_s : u"off"_s;
+    builder.emplaceSystemTextAndUpdate(op, text);
+    builder.emplaceSystemTextAndUpdate(mode, text);
+    builder.emplaceSystemTextAndUpdate(u"mode."_s, text);
+
+    if (!duration.isEmpty())
+    {
+        builder.emplaceSystemTextAndUpdate(duration, text);
+    }
```

Comment:
Followers-only mode can have a "0 minute" duration, we should omit the duration text if that's the case.

![image](https://github.com/user-attachments/assets/ad990a31-fdec-42d1-93ba-dd1819e60731)

The `followers` object looks like this:

```json
    "followers": {
      "follow_duration_minutes": 0
    },
```